modules/dns_common/dns_common.rb

```diff
@@ -24,5 +27,43 @@ def dns_find key
     rescue Resolv::ResolvError
       false
     end
+
+    def ptr_to_ip ptr
+     if ptr =~ /\.in-addr\.arpa$/
+       ptr.split('.')[0..-3].reverse.join('.')
+     elsif ptr =~ /\.ip6\.arpa$/
+       ptr.split('.')[0..-3].reverse.each_slice(4).inject([]) {|address, word| address << word.join}.join(":")
+     else
+       raise Proxy::Dns::Error.new("Not a PTR record: '#{ptr}'")
+     end
+    end
+
+    # conflict methods return values:
+    # no conflict: -1; conflict: 1, conflict but record / ip matches: 0
+    def a_record_conflicts(fqdn, ip)
+      if ip_addr = to_ipaddress(ip)
+        addresses = resolver.getaddresses(fqdn).reject { |a| a =~ Resolv::IPv6::Regex }
```

Comment:
Sorry, should have been more clear. Using `#collent` with IPv4 regex makes more sense to me also; though I currently see no issues with your code since `#getaddesses` only gets us A or AAAA records.